modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/CogConfigurationBean.java

```diff
@@ -0,0 +1,145 @@
+public class CogConfigurationBean {
+
+    public CogConfigurationBean() {};
+
+    public CogConfigurationBean(final CogConfigurationBean that) {
+        Utilities.ensureNonNull("CogConfigurationBean", that);
+        try {
+            BeanUtils.copyProperties(this, that);
+        } catch (IllegalAccessException e) {
+            final IllegalArgumentException iae = new IllegalArgumentException(e);
+            throw iae;
+        } catch (InvocationTargetException e) {
+            final IllegalArgumentException iae = new IllegalArgumentException(e);
+            throw iae;
+        }
+    }
+
+    public CogConfigurationBean(final Indexer indexer) {
+        String cogRangeReader = IndexerUtils.getParameter(Utils.Prop.COG_RANGE_READER, indexer);
+        if (cogRangeReader != null) {
+            this.rangeReader = cogRangeReader;
+        }
+        String user = IndexerUtils.getParameter(Utils.Prop.COG_USER, indexer);
+        if (user != null) {
+            this.user = user;
+        }
+        String password = IndexerUtils.getParameter(Utils.Prop.COG_PASSWORD, indexer);
+        if (password != null) {
+            this.password = password;
+        }
+        if (IndexerUtils.getParameterAsBoolean(Utils.Prop.COG_USE_CACHE, indexer)) {
+            this.useCache = true;
+        }
+    }
+
+    /** Optional user/accessId to access the data */
+    private String user;
+
+    /** Optional password/secret key to access the data */
+    private String password;
+
+    /** Full classname of the rangeReader implementation being used */
+    private String rangeReader;
+
+    /** Whether to use Caching stream or not */
+    private boolean useCache;
+
+    public String getUser() {
+        return user;
+    }
+
+    public void setUser(String user) {
+        this.user = user;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getRangeReader() {
+        return rangeReader;
+    }
+
+    public void setRangeReader(String rangeReader) {
+        this.rangeReader = rangeReader;
+    }
+
+    public boolean isUseCache() {
+        return useCache;
+    }
+
+    public void setUseCache(boolean useCache) {
+        this.useCache = useCache;
+    }
+
+    /**
+     * Create a {@link CogUri} instance on top of the provided url String optionally embedding (when
+     * provided) user/pass credentials as UserInfo component of the underlying URI.
+     */
+    public CogUri createUri(String url) {
+        // Create basic uri
+        URI uri = URI.create(url);
+        String userInfo = uri.getUserInfo();
+        if (userInfo == null) {
+            // Add credentials to URI when provided
+            String user = getUser();
+            String password = getPassword();
+            if (user != null && password != null) {
+                userInfo = user + ":" + password;
```

Comment:
Wondering what would happen with S3 or Azure though? I guess this is for standard HTTP. However, putting the username and password in the URL does not seem awfully secure, what if they end up in some logs for example?

Reply:
@aaime 
This is how both the HTTP and the S3 COG implementation works on imageio-ext. 
They are getting the user and pass from the URI.
What about having all the Imageio-ext classes currently dealing with URI be updated to directly use CogUri instead so that they can get user and pass directly from there?

Reply:
Could be an option, but then it would not be any more a "COG" URI, but hum... something else, more general. BasicAuthURI maybe?

Reply:
@aaime The CogUri also has caching flag info though
https://github.com/geosolutions-it/imageio-ext/blob/master/plugin/cog/cog-commons/src/main/java/it/geosolutions/imageioimpl/plugins/cog/CogUri.java

What about CogAuthUri?
BTW, user and pass might be null

Reply:
Updated as BasicAuthURI.